### PR TITLE
Support for GoCmd library, new command go-recursive-publish and remove recursive flags

### PR DESCRIPTION
**Pull request by AlexeiVainshtein**

### CODE REVIEW

File: jfrog-cli/docs/artifactory/gorecursivepublish/help.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package gorecursivepublish
+
+const Description = "Populates and publish dependencies and transitive dependencies of the project"
```
Comment (eyalbe4) [inline jfrog-cli/docs/artifactory/gorecursivepublish/help.go]: I suggest to rephrase the description to:
"Recursively discovers all project dependencies, both direct and indirect, and publishes them to Artifactory."

File: jfrog-cli/jfrog/buildtools_test.go [hunk 2]
```diff
@@ -499,6 +659,24 @@ func validateBuildInfoProperties(buildInfo buildinfo.BuildInfo, t *testing.T) {
...
+
+		value, contains = propertiesMap["go.version"]
+		if !contains {
+			t.Error("The go.version property is missing on", item.Name)
+		}
+
+		if value == "" {
+			t.Error("The go.version value is empty for", item.Name)
+		}
+
```
Comment (eyalbe4) [inline jfrog-cli/jfrog/buildtools_test.go]: Redundant new-line here.

**Comment by eyalbe4** [comment]
Great work @AlexeiVainshtein!!!
See my inline suggestions.

File: jfrog-cli/artifactory/commands/golang/go.go [hunk 3]
```diff
@@ -1,127 +1,33 @@
...
-			if err != nil {
-				return err
-			}
-		} else {
-			return err
-		}
+		return err
 	}
 
+	err = GoCmd.RunWithFallbacksAndPublish(goArg, targetRepo, noRegistry, serviceManager)
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply jfrog-cli/artifactory/commands/golang/go.go]: Done

[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply jfrog-cli/artifactory/commands/golang/go.go]: Add check for the err.

File: jfrog-cli/artifactory/commands/golang/gorecursivepublish.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+	"strings"
+)
+
+func Execute(targetRepo, goModEditMessage string, serviceManager *artifactory.ArtifactoryServicesManager) error {
+	modFileExists, err := fileutils.IsFileExists("go.mod", false)
+	if err != nil {
+		return err
+	}
+	gmi := goModInfo{}
+	wd, err := os.Getwd()
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply jfrog-cli/artifactory/commands/golang/gorecursivepublish.go]: Done

File: jfrog-cli/docs/artifactory/gorecursivepublish/help.go (hunk 1 above)
Comment (AlexeiVainshtein) [reply jfrog-cli/docs/artifactory/gorecursivepublish/help.go]: Done

File: jfrog-cli/jfrog/buildtools_test.go (hunk 2 above)
Comment (AlexeiVainshtein) [reply jfrog-cli/jfrog/buildtools_test.go]: Done

File: jfrog-cli/artifactory/cli.go [hunk 5]
```diff
@@ -1377,25 +1388,56 @@ func goCmd(c *cli.Context) {
...
+	if c.NArg() != 1 {
+		cliutils.PrintHelpAndExitWithError("Wrong number of arguments.", c)
+	}
+
+	targetRepo := c.Args().Get(0)
+	if targetRepo == "" {
+		cliutils.PrintHelpAndExitWithError("Missing target repo.", c)
+	}
+	details := createArtifactoryDetailsByFlags(c, true)
+	serviceManager, err := utils.CreateServiceManager(details, false)
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply jfrog-cli/artifactory/cli.go]: Done

**Pull request merged**